server/src/main/kotlin/core/pokerhands/Straight.kt

```diff
@@ -0,0 +1,29 @@
+package core.pokerhands
+
+import core.Card
+import core.CardRank
+
+
+class Straight(cards: List<Card>) : PokerHand(HandRank.STRAIGHT, cards) {
+
+    override fun isValidHand(): Boolean = isStraight(cards)
+
+    override fun compareWithinRank(other: PokerHand): Int {
+        val thisRanks = this.cards.map{ it.rank }.toSet()
+        val otherRanks = other.cards.map{ it.rank }.toSet()
+
+        val thisHighestRank = getHighestRank(thisRanks)
+        val otherHighestRank = getHighestRank(otherRanks)
+
+        return thisHighestRank.strength - otherHighestRank.strength
+    }
+
+    private fun getHighestRank(ranks: Set<CardRank>): CardRank {
+        val lowestStraight = (CardRank.TWO..CardRank.FIVE).toSet() + CardRank.ACE
```

Comment:
This is a constant, I'd make it static.

Reply:
It is, its a „val”

Reply:
I moved the functionality to handUtils, maybe we should consider implementing some sort of compareStraight function in utils to avoid code duplication in Straight and StraightFlush classes.